packages/hyperion-flowlet/src/TriggerFlowlet.ts

```diff
@@ -0,0 +1,30 @@
+/**
+ * Copyright (c) Meta Platforms, Inc. and affiliates. All Rights Reserved.
+ */
+
+import { Flowlet, FlowletDataType } from "./Flowlet";
+
+interface TriggerFlowletPayload extends Object {
+  __ext_triggerFlowlet?: TriggerFlowlet
+}
+
+// export class TriggerFlowlet extends Flowlet<FlowletDataType> {
+//   constructor(
+//     name: string,
+//     parent?: Flowlet | null
+//   ) {
+//     super(name, parent);
+//     this.data.triggerFlowlet = this; // Ensure this flowlet report itself as the trigger.
+//   }
+// }
+
+export interface TriggerFlowlet extends Flowlet<FlowletDataType> { }
+
+export function setTriggerFlowlet(obj: TriggerFlowletPayload, triggerFlowlet: TriggerFlowlet): void {
+  triggerFlowlet.data.triggerFlowlet = triggerFlowlet;
+  obj.__ext_triggerFlowlet = triggerFlowlet;
```

Comment:
Elsewhere in Hyperion, I've seen you use `setVirtualPropertyValue`. Why not use that here as well?

Reply:
I started with that, but then I realized we want to use this same function to also assign triggerFlowlet to normal objects, such as action payload, and graphapi params. 
`setVirtualPropertyValue` requires the object to be interceptable. 
So, for now I'm hiding the actual implementation behind these two function (set/getTriggerFlowlet). If we decided to allow normal objects be intercepted (e.g. add IObject to core), then we can switch the implementation.